### DISCUSSION TITLE
help!:)Type error: No overload matches this call.   Overload 2 of 2, '(requesters: Requester): Redis', gave the following error.     Argument of type '{ url: string | undefined; token: string | undefined; }' is not assignable to parameter of type 'Requester'.       Object literal may only specify known properties, and 'url' does not exist in type 'Requester'. error Command failed with exit code 1. info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command. Error: Command "yarn run build" exited with 1 Deployment completed BUILD_UTILS_SPAWN_1: Command "yarn run build" exited with 1

### DISCUSSION THREAD

**Discussion by zixgit** [vercel/next.js#48737] [Help]
### Summary

<html>
<body>

Type error: No overload matches this call.
--
20:36:26.810 | Overload 2 of 2, '(requesters: Requester): Redis', gave the following error.
20:36:26.810 | Argument of type '{ url: string \| undefined; token: string \| undefined; }' is not assignable to parameter of type 'Requester'.

20:36:26.810 |  const redis = new Redis({
20:36:26.810 | >  9 \|   url: process.env.REDIS_URL,
20:36:26.810 | \|   ^
20:36:26.810 | 10 \|   token: process.env.REDIS_SECRET,
20:36:26.810 | 11 \| })

20:36:26.810 | Object literal may only specify known properties, and 'url' does not exist in type 'Requester'.
Expand 8 Lines
20:36:26.945 | error Command failed with exit code 1.
20:36:26.946 | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
20:36:26.963 | Error: Command "yarn run build" exited with 1
20:36:27.221 | Deployment completed
20:36:27.181 | BUILD_UTILS_SPAWN_1: Command "yarn run build" exited with 1

</body>
</html>
![image](https://user-images.githubusercontent.com/110596981/233820362-b3b51d24-e466-4a37-bdc6-3e4975ab1ca1.png)

**Comment by icyJoseph** [comment]
Hi, 

This is a type error from the Redis TypeScript definitions. 

How are you seeing this error? During dev time? during deploy? 

What the error is saying is that there's no matching shape for the matching the presence of a `url`, in the Requester object. I am trying to find the type definitions, to see where you are going wrong, but without further context is a bit hard to help.

What Redis client/package are you using?

**Comment by zixgit** [comment]
Hello,I send this email to tell you I have success to deploy!
To deal with last email mentioned problem when deploying,I guess because of Redis version difference cause the error ,and I use code as follows to replace:
const redis = new Redis({
4  url: 'https://XXXXXX.upstash.io', 5  token: '********',&nbsp;6})&nbsp;
I appreciate your video ,which let me deploy first website.Thank you and your reply!




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "vercel/next.js"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年4月24日(星期一) 下午3:24
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [vercel/next.js] help!:)Type error: No overload matches this call.   Overload 2 of 2, '(requesters: Requester): Redis', gave the following error.     Argument of type '{ url: string | undefined; token: string | undefined; }' is not assignable to parameter of type 'Requester'.       Object literal may only specify known properties, and 'url' does not exist in type 'Requester'. error Command failed with exit code 1. info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command. Error: Command "yarn run build" exited with 1 Deployment completed BUILD_UTILS_SPAWN_1: Command "yarn run build" exited with 1 (Discussion #48737)





 
Hi,
 
This is a type error from the Redis TypeScript definitions.
 
How are you seeing this error? During dev time? during deploy?
 
What the error is saying is that there's no matching shape for the matching the presence of a url, in the Requester object. I am trying to find the type definitions, to see where you are going wrong, but without further context is a bit hard to help.

**Reply by QiYongchuan** [reply]
Hi, I  meet the same question and see your this answer, by this way, i resolved my bug. Thanks you!   And i have other things want to discuss with  you, in this method, is there a risk of token leakage? After all, the token is directly placed in the code.

**Reply by zixgit** [reply]
You are welcome:)
As for your worry about token leakage, I find from web and think set a exp time is a way to desolve it.

**Comment by davidlwg** [comment]
Found this discussion while getting the same error. Fixed it after seeing another workaround, posted below in case it helps anybody else.

```Javascript
export const db = Redis.fromEnv()
```

Source: https://stackoverflow.com/a/76118438/21908083

**Reply by ismumchow** [reply]
would you do this instead of export const db = prisma?
